sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_document_retrieval/_document_retrieval.py

```diff
@@ -0,0 +1,342 @@
+        ideal_docs_groundtruth_labels = [label for (_, label) in qrels_sorted_by_rank]
+
+        # calculate the proportion of result docs with no ground truth label (holes)
+        holes = self._compute_holes(
+            [x[0] for x in results_sorted_by_rank], [x[0] for x in qrels_sorted_by_rank]
+        )
+        holes_ratio = holes / float(len(results))
+
+        # if none of the retrieved docs are labeled, report holes only
+        if not any(result_docs_groundtruth_labels):
+            metrics = {
+                f"ndcg@{self.k}": 0,
+                f"xdcg@{self.k}": 0,
+                "fidelity": 0,
+                "top1_relevance": 0,
+                "top3_max_relevance": 0,
+                "holes": holes,
+                "holes_ratio": holes_ratio,
+                "total_retrieved_documents": len(results),
+                "total_groundtruth_documents": len(qrels),
+            }
+            binary_result = self._get_binary_result(**metrics)
+            for k, v in binary_result.items():
+                metrics[k] = v
+
+            return metrics
+
+        metrics = {
+            f"ndcg@{self.k}": self._compute_ndcg(
+                result_docs_groundtruth_labels[: self.k],
+                ideal_docs_groundtruth_labels[: self.k],
+            ),
+            f"xdcg@{self.k}": self._compute_xdcg(
+                result_docs_groundtruth_labels[: self.k]
+            ),
+            "fidelity": self._compute_fidelity(
+                result_docs_groundtruth_labels, ideal_docs_groundtruth_labels
+            ),
+            "top1_relevance": result_docs_groundtruth_labels[0],
+            "top3_max_relevance": max(result_docs_groundtruth_labels[: self.k]),
+            "holes": holes,
+            "holes_ratio": holes_ratio,
+            "total_retrieved_documents": len(results),
+            "total_groundtruth_documents": len(qrels),
+        }
+
+        binary_result = self._get_binary_result(**metrics)
+        for k, v in binary_result.items():
+            metrics[k] = v
+
+        return metrics
+
+    @overload
+    def __call__(  # type: ignore
+        self,
+        *,
+        retrieval_ground_truth: List[RetrievalGroundTruthDocument],
+        retrieved_documents: List[RetrievedDocument],
+    ) -> DocumentRetrievalMetrics:
+        """
+        Compute document retrieval metrics for documents retrieved from a search algorithm against a known set of ground truth documents.
+
+        Input `retrieval_ground_truth` is an instance of `List[azure.ai.evaluation.RetrievalGroundTruthDocument]`, where each item of the list represents a ground-truth judgement for a particular document relative to a query.
+        Input `retrieved_documents` is an instance of `List[azure.ai.evaluation.RetrievedDocument]`, where each item of the list represents a document scored by a search algorithm for the same query.
+
+        Evaluation metrics calculated include NDCG@3, XDCG@3, Fidelity, Top K Relevance and Holes.
```

Comment:
These doc string might now render as expected with MS docs. Please take a look at existing evaluator for reference and update it.

Reply:
I have updated all of the docstrings to match what I found in other evaluators.  Let me know if there is anything I've missed or if you have other suggestions.